trunk/src/app/srs_app_http_stream.cpp

```diff
@@ -1032,13 +1056,40 @@ srs_error_t SrsHttpStreamServer::http_mount(SrsRequest* r)
 void SrsHttpStreamServer::http_unmount(SrsRequest* r)
 {
     std::string sid = r->get_stream_url();
-    
-    if (sflvs.find(sid) == sflvs.end()) {
+
+    std::map<std::string, SrsLiveEntry*>::iterator it = sflvs.find(sid);
+    if (it == sflvs.end()) {
         return;
     }
-    
-    SrsLiveEntry* entry = sflvs[sid];
-    entry->stream->entry->enabled = false;
+
+    // Free all HTTP resources.
+    SrsLiveEntry* entry = it->second;
+    SrsAutoFree(SrsLiveEntry, entry);
+    sflvs.erase(it);
+
+    SrsLiveStream* stream = entry->stream;
+    SrsAutoFree(SrsLiveStream, stream);
```

Comment:
the `SrsAutoFree` not good enough here, when `stream` being released, `stream = NULL`, but the entry->stream still hold an not NULL pointer, but it's memory already released.

So is it a good idea to let the `SrsLiveEntry` destructor to free `stream`, `cache`?

Reply:
I think it's ok, because the entry is removed and freed:

```cpp
    SrsLiveEntry* entry = it->second;
    SrsAutoFree(SrsLiveEntry, entry);
    streamHandlers.erase(it);

    SrsLiveStream* stream = entry->stream;
    SrsAutoFree(SrsLiveStream, stream);

    SrsBufferCache* cache = entry->cache;
    SrsAutoFree(SrsBufferCache, cache);
```